dallinger/heroku/tools.py

```diff
@@ -418,7 +418,8 @@ def __enter__(self):
         return self
 
     def __exit__(self, exctype, excinst, exctb):
-        self.stop()
+        if self.is_running:
```

Comment:
`stop()` already checks itself; do we need to check twice? (This would be clearer if `stop()` used `is_running` to check, instead of repeating the `self._process is None` check.)

Reply:
When the status-monitoring thread determines that the experiment is complete, it calls `heroku.stop()` to make sure that the output stream ends even if there's no more output from the heroku subprocesses. But the heroku wrapper is also a context manager (in `LocalSessionRunner.run`) so when it exits it calls `heroku.stop()` again, and I wanted to avoid printing the "No local Heroku process was running." message in this case.

Reply:
Ah, yes... makes sense.